hack/lint-dependencies.sh

```diff
@@ -41,19 +44,22 @@ kube::util::require-jq
 # let us log all errors before we exit
 rc=0
 
+# Find binary
+dependencyverifier=$(kube::util::find-binary "dependencyverifier")
+
+if [[ ! -x "$dependencyverifier" ]]; then
+  {
+    echo "It looks as if you don't have a compiled dependencyverifier binary"
+    echo
+    echo "If you are running from a clone of the git repo, please run"
+    echo "'make WHAT=cmd/dependencyverifier'."
+  } >&2
+  exit 1
+fi
+
 # List of dependencies we need to avoid dragging back into kubernetes/kubernetes
-forbidden_repos=(
-  "k8s.io/klog"  # we have switched to klog v2, so avoid klog v1
-)
-for forbidden_repo in "${forbidden_repos[@]}"; do
-  deps_on_forbidden=$(go mod graph | grep " ${forbidden_repo}@" || echo "")
-  if [ -n "${deps_on_forbidden}" ]; then
-    kube::log::error "The following have transitive dependencies on ${forbidden_repo}, which is not allowed:"
-    echo "${deps_on_forbidden}"
-    echo ""
-    rc=1
-  fi
-done
+# Check if unwanted dependencies are removed
+"${dependencyverifier}" "${KUBE_ROOT}/hack/unwanted-dependencies.json"
```

Comment:
suggest simplifying this to

```
go run k8s.io/kubernetes/cmd/dependencyverifier "${KUBE_ROOT}/hack/unwanted-dependencies.json"
```

then you don't need to make, find-binary, and verify it was found

Reply:
this is really noisy in the success case... running this, I get the following:

> `$ hack/lint-dependencies.sh `
> `Mod Graph: `
> ` -- map[cloud.google.com/go:cloud.google.com/go/datastore cloud.google.com/go/bigquery:cloud.google.com/go cloud.google.com/go/datastore:cloud.google.com/go cloud.google.com/go/firestore:google.golang.org/grpc cloud.google.com/go/pubsub:cloud.google.com/go cloud.google.com/go/storage:google.golang.org/grpc github.com/Azure/go-autorest/autorest:github.com/Azure/go-autorest/autorest/adal github.com/Azure/go-autorest/autorest/adal:github.com/Azure/go-autorest github.com/Azure/go-autorest/autorest/date:github.com/Azure/go-autorest github.com/Azure/go-autorest/autorest/mocks:github.com/Azure/go-autorest github.com/Azure/go-autorest/autorest/validation:github.com/stretchr/testify github.com/Azure/go-autorest/logger:github.com/Azure/go-autorest github.com/Azure/go-autorest/tracing:github.com/Azure/go-autorest github.com/GoogleCloudPlatform/k8s-cloud-provider:cloud.google.com/go github.com/Microsoft/go-winio:github.com/pkg/errors github.com/Microsoft/hcsshim:github.com/Microsoft/go-winio github.com/NYTimes/gziphandler:github.com/stretchr/testify github.com/alecthomas/units:github.com/stretchr/testify github.com/aws/aws-sdk-go:github.com/jmespath/go-jmespath github.com/bketelsen/crypt:google.golang.org/grpc github.com/checkpoint-restore/go-criu/v5:google.golang.org/protobuf github.com/cilium/ebpf:golang.org/x/sys github.com/cncf/udpa/go:google.golang.org/grpc github.com/cockroachdb/datadriven:github.com/certifi/gocertifi github.com/containerd/cgroups:golang.org/x/sys github.com/containerd/console:golang.org/x/sys github.com/containerd/ttrpc:google.golang.org/grpc github.com/containerd/typeurl:github.com/pkg/errors github.com/coredns/caddy:github.com/google/uuid github.com/coredns/corefile-migration:github.com/coredns/caddy github.com/coreos/go-systemd/v22:github.com/godbus/dbus/v5 github.com/cpuguy83/go-md2man/v2:github.com/shurcooL/sanitized_anchor_name github.com/envoyproxy/go-control-plane:google.golang.org/protobuf github.com/frankban/quicktest:github.com/kr/pretty github.com/fsnotify/fsnotify:golang.org/x/sys github.com/go-kit/log:github.com/go-stack/stack github.com/go-openapi/jsonpointer:github.com/stretchr/testify github.com/go-openapi/jsonreference:golang.org/x/text github.com/go-openapi/swag:gopkg.in/yaml.v2 github.com/gogo/protobuf:golang.org/x/tools github.com/golang/mock:rsc.io/quote/v3 github.com/golang/protobuf:google.golang.org/protobuf github.com/google/cadvisor:cloud.google.com/go github.com/google/go-cmp:golang.org/x/xerrors github.com/google/pprof:golang.org/x/sys github.com/googleapis/gax-go/v2:google.golang.org/grpc github.com/googleapis/gnostic:github.com/docopt/docopt-go github.com/gophercloud/gophercloud:gopkg.in/yaml.v2 github.com/grpc-ecosystem/go-grpc-middleware:google.golang.org/grpc github.com/grpc-ecosystem/grpc-gateway:gopkg.in/yaml.v2 github.com/hashicorp/consul/api:github.com/hashicorp/consul/sdk github.com/hashicorp/consul/sdk:github.com/hashicorp/go-cleanhttp github.com/hashicorp/go-immutable-radix:github.com/hashicorp/go-uuid github.com/hashicorp/go-multierror:github.com/hashicorp/errwrap github.com/hashicorp/go-rootcerts:github.com/mitchellh/go-homedir github.com/hashicorp/hcl:github.com/davecgh/go-spew github.com/hashicorp/mdns:github.com/hashicorp/go.net github.com/hashicorp/memberlist:github.com/armon/go-metrics github.com/hashicorp/serf:github.com/armon/circbuf github.com/jmespath/go-jmespath:github.com/jmespath/go-jmespath/internal/testify github.com/jmespath/go-jmespath/internal/testify:gopkg.in/yaml.v2 github.com/json-iterator/go:github.com/stretchr/testify github.com/kisielk/errcheck:golang.org/x/tools github.com/kr/pretty:github.com/kr/text github.com/kr/text:github.com/creack/pty github.com/matttproud/golang_protobuf_extensions:golang.org/x/sync github.com/miekg/dns:golang.org/x/crypto github.com/mitchellh/cli:github.com/armon/go-radix github.com/moby/ipvs:github.com/pkg/errors github.com/moby/spdystream:github.com/gorilla/websocket github.com/moby/sys/mountinfo:golang.org/x/sys github.com/moby/term:gotest.tools/v3 github.com/niemeyer/pretty:github.com/kr/text github.com/nxadm/tail:gopkg.in/tomb.v1 github.com/olekukonko/tablewriter:github.com/mattn/go-runewidth github.com/onsi/ginkgo:golang.org/x/text github.com/onsi/gomega:gopkg.in/yaml.v2 github.com/opencontainers/runc:github.com/checkpoint-restore/go-criu/v5 github.com/opencontainers/selinux:github.com/pkg/errors github.com/prometheus/client_golang:google.golang.org/protobuf github.com/prometheus/client_model:golang.org/x/sync github.com/prometheus/common:gopkg.in/yaml.v2 github.com/prometheus/procfs:golang.org/x/sys github.com/rogpeppe/go-internal:gopkg.in/errgo.v2 github.com/sergi/go-diff:github.com/davecgh/go-spew github.com/sirupsen/logrus:golang.org/x/sys github.com/smartystreets/goconvey:golang.org/x/tools github.com/soheilhy/cmux:golang.org/x/net github.com/spf13/afero:golang.org/x/text github.com/spf13/cast:github.com/stretchr/testify github.com/spf13/cobra:gopkg.in/yaml.v2 github.com/spf13/viper:gopkg.in/yaml.v2 github.com/stoewer/go-strcase:github.com/stretchr/testify github.com/stretchr/objx:github.com/stretchr/testify github.com/stretchr/testify:gopkg.in/yaml.v3 github.com/urfave/cli:gopkg.in/yaml.v2 github.com/vishvananda/netlink:github.com/vishvananda/netns github.com/vishvananda/netns:golang.org/x/sys go.etcd.io/bbolt:golang.org/x/sys go.etcd.io/etcd/api/v3:google.golang.org/grpc go.etcd.io/etcd/client/pkg/v3:github.com/coreos/go-systemd/v22 go.etcd.io/etcd/client/v2:go.etcd.io/etcd/client/pkg/v3 go.etcd.io/etcd/client/v3:github.com/dustin/go-humanize go.etcd.io/etcd/etcdutl/v3:go.uber.org/zap go.etcd.io/etcd/pkg/v3:google.golang.org/grpc go.etcd.io/etcd/raft/v3:go.etcd.io/etcd/client/pkg/v3 go.etcd.io/etcd/server/v3:sigs.k8s.io/yaml go.etcd.io/etcd/tests/v3:gopkg.in/yaml.v2 go.opencensus.io:google.golang.org/grpc go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc:github.com/golang/protobuf go.opentelemetry.io/otel:github.com/google/go-cmp go.opentelemetry.io/otel/exporters/otlp:github.com/google/go-cmp go.opentelemetry.io/otel/metric:github.com/google/go-cmp go.opentelemetry.io/otel/oteltest:go.opentelemetry.io/otel go.opentelemetry.io/otel/sdk:github.com/google/go-cmp go.opentelemetry.io/otel/sdk/export/metric:github.com/stretchr/testify go.opentelemetry.io/otel/sdk/metric:github.com/benbjohnson/clock go.opentelemetry.io/otel/trace:github.com/google/go-cmp go.opentelemetry.io/proto/otlp:github.com/golang/protobuf go.starlark.net:github.com/chzyer/logex go.uber.org/atomic:github.com/stretchr/testify go.uber.org/goleak:github.com/kr/pretty go.uber.org/multierr:go.uber.org/atomic go.uber.org/zap:gopkg.in/yaml.v3 golang.org/x/crypto:golang.org/x/term golang.org/x/exp:golang.org/x/xerrors golang.org/x/image:golang.org/x/text golang.org/x/lint:golang.org/x/tools golang.org/x/mobile:golang.org/x/tools golang.org/x/mod:golang.org/x/xerrors golang.org/x/net:golang.org/x/text golang.org/x/oauth2:google.golang.org/appengine golang.org/x/term:golang.org/x/sys golang.org/x/text:golang.org/x/tools golang.org/x/tools:golang.org/x/xerrors gonum.org/v1/gonum:gonum.org/v1/plot gonum.org/v1/netlib:modernc.org/xc gonum.org/v1/plot:rsc.io/pdf google.golang.org/api:cloud.google.com/go google.golang.org/appengine:golang.org/x/text google.golang.org/genproto:google.golang.org/protobuf google.golang.org/grpc:google.golang.org/protobuf google.golang.org/protobuf:github.com/google/go-cmp gopkg.in/errgo.v2:github.com/kr/pretty gopkg.in/yaml.v2:gopkg.in/check.v1 gopkg.in/yaml.v3:gopkg.in/check.v1 gotest.tools/v3:golang.org/x/tools honnef.co/go/tools:golang.org/x/mod k8s.io/api:github.com/gogo/protobuf k8s.io/apiextensions-apiserver:github.com/emicklei/go-restful k8s.io/apimachinery:sigs.k8s.io/yaml k8s.io/apiserver:github.com/google/uuid k8s.io/cli-runtime:github.com/davecgh/go-spew k8s.io/client-go:cloud.google.com/go k8s.io/cloud-provider:github.com/google/go-cmp k8s.io/cluster-bootstrap:github.com/stretchr/testify k8s.io/code-generator:github.com/emicklei/go-restful k8s.io/component-base:github.com/blang/semver k8s.io/component-helpers:github.com/google/go-cmp k8s.io/controller-manager:github.com/spf13/pflag k8s.io/cri-api:github.com/davecgh/go-spew k8s.io/csi-translation-lib:github.com/stretchr/testify k8s.io/gengo:github.com/davecgh/go-spew k8s.io/klog/v2:github.com/go-logr/logr k8s.io/kube-aggregator:github.com/spf13/pflag k8s.io/kube-controller-manager:k8s.io/apimachinery k8s.io/kube-openapi:github.com/NYTimes/gziphandler k8s.io/kube-proxy:k8s.io/apimachinery k8s.io/kube-scheduler:github.com/google/go-cmp k8s.io/kubectl:github.com/MakeNowJust/heredoc k8s.io/kubelet:github.com/gogo/protobuf k8s.io/kubernetes:sigs.k8s.io/yaml k8s.io/legacy-cloud-providers:cloud.google.com/go k8s.io/metrics:github.com/gogo/protobuf k8s.io/mount-utils:github.com/kr/text k8s.io/sample-apiserver:github.com/google/gofuzz k8s.io/system-validators:github.com/blang/semver k8s.io/utils:github.com/davecgh/go-spew rsc.io/quote/v3:rsc.io/sampler rsc.io/sampler:golang.org/x/text sigs.k8s.io/apiserver-network-proxy/konnectivity-client:k8s.io/klog/v2 sigs.k8s.io/kustomize/api:k8s.io/kube-openapi sigs.k8s.io/kustomize/cmd/config:k8s.io/kube-openapi sigs.k8s.io/kustomize/kustomize/v4:sigs.k8s.io/yaml sigs.k8s.io/kustomize/kyaml:k8s.io/kube-openapi sigs.k8s.io/structured-merge-diff/v4:gopkg.in/yaml.v2 sigs.k8s.io/yaml:gopkg.in/yaml.v2]`
> `All dependencies:`
> ` -- { `
> `  "forbidden-dependencies": {`
> `    "github.com/go-kit/kit": "",`
> `    "github.com/go-openapi/analysis": "",`
> `    "github.com/go-openapi/spec": "",`
> `    "github.com/go-openapi/strfmt": "",`
> `    "github.com/go-openapi/validate": "",`
> `    "github.com/influxdata/influxdb1-client": "",`
> `    "go.mongodb.org/mongo-driver": "",`
> `    "k8s.io/klog": "we have switched to klog v2, so avoid klog v1",`
> `    "rsc.io/quote": "refer to #102833"`
> `  },`
> `  "unwanted-dependencies": {`
> `    "github.com/hashicorp/hcl": "",`
> `    "github.com/spf13/viper": "refer to #102598",`
> `    "rsc.io/sampler": "refer to #102833"`
> `  }`
> `}`
> 
> `Forbidden dependencies:`
> ` -- map[github.com/go-kit/kit: github.com/go-openapi/analysis: github.com/go-openapi/spec: github.com/go-openapi/strfmt: github.com/go-openapi/validate: github.com/influxdata/influxdb1-client: go.mongodb.org/mongo-driver: k8s.io/klog:we have switched to klog v2, so avoid klog v1 rsc.io/quote:refer to #102833]`
> `Unwanted dependencies:`
> ` -- map[github.com/hashicorp/hcl: github.com/spf13/viper:refer to #102598 rsc.io/sampler:refer to #102833]`
> `2021/06/29 14:10:14 - Check forbidden depencies rsc.io/quote...`
> `2021/06/29 14:10:14 - Check forbidden depencies github.com/go-kit/kit...`
> `2021/06/29 14:10:14 - Check forbidden depencies github.com/go-openapi/spec...`
> `2021/06/29 14:10:14 - Check forbidden depencies github.com/influxdata/influxdb1-client...`
> `2021/06/29 14:10:14 - Check forbidden depencies k8s.io/klog...`
> `2021/06/29 14:10:14 - Check forbidden depencies github.com/go-openapi/analysis...`
> `2021/06/29 14:10:14 - Check forbidden depencies github.com/go-openapi/strfmt...`
> `2021/06/29 14:10:14 - Check forbidden depencies github.com/go-openapi/validate...`
> `2021/06/29 14:10:14 - Check forbidden depencies go.mongodb.org/mongo-driver...`
> `2021/06/29 14:10:14 - Check unwanted depencies github.com/hashicorp/hcl...`
> `2021/06/29 14:10:14 - Check unwanted depencies github.com/spf13/viper...`
> `2021/06/29 14:10:14 - Check unwanted depencies rsc.io/sampler...`
> 
> `real	0m13.041s`
> `user	0m1.134s`
> `sys	0m2.557s`
> `All pinned versions of checked dependencies match their preferred version.`
> 
> `$ echo $?`
> `0`

That looks a lot like an error, not a successful run

Reply:
I removed some debug messages.

Updated.